makelib/golang.mk

```diff
@@ -18,7 +18,7 @@
 # Optional. The Go Binary to use
 GO ?= go
 
-# Optional. Minimum Go version.
+# Optional. Required Go version.
```

Comment:
Is it really the 'Required' version? What about if someone is using 1.19, etc (once those are released)?

Reply:
`GO_REQUIRED_VERSION=1.19 make ...` :D

I just changed the comment and error message because I was not really checking for `GO_VERSION < GO_REQUIRED_VERSION` but rather `GO_REQUIRED_VERSION == GO_VERSION`

Reply:
>I just changed the comment and error message because I was not really checking for GO_VERSION < GO_REQUIRED_VERSION but rather GO_REQUIRED_VERSION == GO_VERSION

Gotcha. What're your thoughts on changing the check to this?:
```
GO_REQUIRED_VERSION <= GO_VERSION
```
That way we don't _have to_ update the build submodule each time we upgrade our go version.

Reply:
`GO_REQUIRED_VERSION` can be set per project: https://github.com/upbound/distro/blob/main/Makefile#L65

So we can change Go versions without needing to change the submodule.